### ISSUE TITLE
Feature Request: Add Rule for Block Consistency in route_type Values Across block_id

### ISSUE THREAD

**Issue by YuliaTom** [MobilityData/gtfs-validator#1925]
### Describe the problem

This is a proposal for a new validation rule for the GTFS Validator, inspired by the GTFS specification's description of `block_id`. According to the spec, `block_id` is used to group trips operated by the same vehicle, and this implies that all trips within the same block should share the same mode of transportation. A validation rule can help ensure that vehicles are not incorrectly switching modes within a block.
Specifically, this rule would check that all trips sharing the same `block_id` in trips.txt reference routes with consistent `route_type` values in routes.txt. If a `block_id` is found to span multiple `route_type` values, a WARNING should be issued.
This enhancement aligns with how some GTFS consumers (such as Google, please see the attached screenshot) handle blocks to ensure feed quality and could prevent potential errors in downstream systems.

*Proposed Rule*
Rule Name: Inconsistent route_type Values in Trips Sharing a `block_id`
Description: Trips with the same `block_id` in the `trips.txt` file must have consistent `route_type` values in the `routes.txt` file. If trips within a block have differing `route_type` values, a warning needs to be issued.
Severity: WARNING
Affected Files:
`trips.txt`
`routes.txt`

### Describe the new validation rule

 routeTypeMap = {route["route_id"]: route["route_type"] for route in gtfsData["routes.txt"]}

    blockRouteTypes = {}
    for trip in gtfsData["trips.txt"]:
        block_id = trip["block_id"]
        route_id = trip["route_id"]
        if block_id and route_id in routeTypeMap:
            blockRouteTypes.setdefault(block_id, set()).add(routeTypeMap[route_id])

    warnings = []
    for block_id, route_types in blockRouteTypes.items():
        if len(route_types) > 1:
            warnings.append(f"Inconsistent route_types for block_id '{block_id}': {list(route_types)}")

    return warnings

### Sample GTFS datasets

[austin_gtfs.zip](https://github.com/user-attachments/files/17918076/austin_gtfs.zip)

In this GTFS sample block_id 20090-1-161 is associated with routes 20 and 484. Route 20 is of type 1 and 484 of type 3, which should trigger the inconsistent route type warning. 

### Severity

WARNING

### Additional context

![Image](https://github.com/user-attachments/assets/2cc75e81-729b-43b0-94f9-1e183ad45460)

**Comment by emmambd** [comment]
Thanks for flagging this @YuliaTom! We agree this should be added as a warning in the validator and will be added to our backlog. 

If you'd like to accelerate the development of this rule, you are welcome to add it and open a PR for review! [Read more about our code contribution guidelines and process here.](https://github.com/MobilityData/gtfs-validator/blob/master/docs/CONTRIBUTING.md#code-contributions)

**Comment by cka-y** [comment]
### Proposed WARNING level notice: `inconsistent_route_type_for_block_id`
Flags cases where trips sharing the same `block_id` reference routes with different `route_type` values (i.e. one vehicle switching transit modes).

**Suggested Notice Description**  
All trips for a `block_id` should belong to the same transport mode.

**Notice Columns**
| File        | Column       | Suggested column description |
|-------------|--------------|---------|
| `trips.txt` | `trip_id`   | Trip ID of the problematic trip |
| `trips.txt` | `block_id`   | Block ID of the problematic trip |
| `trips.txt` | `route_id`   | Route ID of the problematic trip |
| `routes.txt`| `route_type` | Transit mode |

cc: @skalexch to validate/update the suggestion

**Comment by skalexch** [comment]
@cka-y @emmambd Agreed on the format. In fact, I would go further and have another test for this rule:
In transfers.txt, if transfer_type=4 (in-seat transfer), then `route_type(from_route_id) = route_type(to_route_id)`. Which unfortunately might have a different notice name and description.

Additional note: We would want to revisit this if at any point in the future we expand route types so that buses include multiple values.

Let me know if there's anything to change in the table.

### Notice Requirements

Related spec language | Rationale for rule | Notice name to update or add | Pseudologic | Notice table columns | Severity | Notice description
-- | -- | -- | -- | -- | -- | --
trips.block_id, trips.trip_id, trips.route_id, routes.route_type | A block should have the same route mode. | inconsistent_route_type_for_block_id| For each trip_id: <br> &nbsp; Get block_id and route_id <br> Create Map<block_id, List[route_type(route_id)]> <br><br> For each block_id in BlockIdMap: <br> &nbsp; If len( List[route_type(route_id)]) > 1: Trigger| blockId, routeIds (list of route_ids in the block) | WARNING | All trips for a `block_id` should belong to routes of the same transport mode.
tranfers.transfer_type, transfers.from_route_id, transfers.to_route_id, routes.route_type | An in-seat transfer should occur in the same route mode.| inconsistent_route_type_for_in_seat_transfer | For each transfer, if transfer_type=4: <br><br> If route_type(from_route_id) != route_type (to_route_id): Trigger | csvRowNumber (entry in transfers.txt), fromRouteId, fromRouteType, toRouteType | WARNING | All routes for an in-seat transfer should have the same transport mode.

**Comment by cka-y** [comment]
Thanks @skalexch! Could you give more details to what you're expecting in the notice columns under fieldname, fieldvalue,...?

**Comment by skalexch** [comment]
@cka-y I fixed it in the table. It makes more sense to just list the `blockId` only, because we do not know how many different routeIds we will have, and we'd want to to raise one notice per `block_id` only. Do you think there's a way to show all those routeIds?

For the transfers.txt case, we have an easier one-entry-per-row, so `csvRowNumber (entry in transfers.txt), fromRouteId, fromRouteType, toRouteType` is simple to retrieve.

**Comment by cka-y** [comment]
@skalexch how about having a comma separated list of routes as routeIds?

**Comment by skalexch** [comment]
Yes that works, I will include it.
